app/src/main/java/com/github/swent/echo/viewmodels/event/EventViewModel.kt

```diff
@@ -29,30 +43,77 @@ class EventViewModel() : ViewModel() {
         )
     private val _event = MutableStateFlow<Event>(emptyEvent)
     private val _status = MutableStateFlow<EventStatus>(EventStatus.New)
+    private val allTagsList = MutableStateFlow<List<Tag>>(listOf())
 
-    // placeholder for the list of all tags
-    private val allTagsList = listOf(Tag("1", "tag1"), Tag("2", "tag2"), Tag("3", "tag3"))
+    // initialize async values
+    init {
+        _status.value = EventStatus.Saving // avoid saving before initialization
+        viewModelScope.launch {
+            val userid = authenticationService.getCurrentUserID()
+            if (userid == null) {
+                _status.value = EventStatus.Error(R.string.event_creation_error_not_logged_in)
+                Log.e("create event", "the user is not logged in")
+            } else {
+                _event.value =
+                    _event.value.copy(
+                        creatorId = userid,
+                        organizerId = userid,
+                        organizerName = repository.getUserProfile(userid).name
+                    )
+                _status.value = EventStatus.New
+            }
+            allTagsList.value = repository.getAllTags()
+        }
+    }
 
-    // return the event
-    fun getEvent(): Event {
-        return _event.asStateFlow().value
+    // constructor for an already existing event
+    constructor(
+        repository: Repository,
+        authenticationService: AuthenticationService,
+        eventId: String
+    ) : this(repository, authenticationService) {
+        _status.value = EventStatus.Saving // avoid saving before initialization
+        viewModelScope.launch {
+            _event.value = repository.getEvent(eventId)
+            _status.value = EventStatus.Saved
+        }
```

Comment:
Unsure how this is going to work. Normally hilt will automatically create the view model and inject the dependencies... When will this constructor be called ? I think the solution is to use see [ViewModel SavedStateHandle](https://developer.android.com/topic/libraries/architecture/viewmodel/viewmodel-savedstate) and [Hilt SavedStateHandle](https://developer.android.com/training/dependency-injection/hilt-jetpack). The `SavedStateHandle` is like a map, you could try to get the `eventId` from it. If it returns `null` you need to create a new event otherwise you modify the event with the given `eventId`. Also, in this case you don't need to get the user id as it will already be saved in the `Event`.